### PR TITLE
Editor leaderboard

### PR DESCRIPTION
**Pull request by Emmanueldmlr**
The idea is to show the top editors for a period of time and their earnings

fixes https://github.com/EveripediaNetwork/issues/issues/613

### CODE REVIEW

**Comment by Emmanueldmlr** [comment]
@kesar, I'm done with the UI design. I want to know if the API is ready.

**Review by Simplypheyie96 — COMMENTED** [review]
why not have other variations like on figma instead of just cesar all through? @Emmanueldmlr

**Review by Simplypheyie96 — CHANGES_REQUESTED** [review]
Also, the editors have different trophies for the !st, 2nd and 3rd editor, the a star for the rest of the editors.
ps: will the implementation of the top editor for user profile be on this pr too? @Emmanueldmlr 
image

**Review by Simplypheyie96 — CHANGES_REQUESTED** [review]
Also, the editors have different trophies for the !st, 2nd and 3rd editor, the a star for the rest of the editors.
ps: will the implementation of the top editor for user profile be on this pr too? @Emmanueldmlr 
image

**Comment by Emmanueldmlr** [comment]
@Simplypheyie96, it's just mock data, they will change once I integrate it with the right endpoint.

**Comment by Emmanueldmlr** [comment]
@OleanjiKingCode, let me know when you are available for a sync up.

**Comment by Emmanueldmlr** [comment]
@OleanjiKingCode, let me know when you are done with the update we discussed.

**Comment by Emmanueldmlr** [comment]
Oh...I already switched to the pending issue on the IQ dashboard.

**Comment by Emmanueldmlr** [comment]
@OleanjiKingCode, what's the status of the ranking and value format we discussed on the call?

**Comment by Emmanueldmlr** [comment]
image

@Simplypheyie96, what we have now

**Comment by Emmanueldmlr** [comment]
@OleanjiKingCode @kesar, I think we will need an endpoint to get the list of editors well ranked having their rank included as part of the object property, and also, another endpoint to fetch a single editor with the rank included. That will take off the sorting I'm doing on the front end and I will be able to display their individual rank on their profile too. @kesar, let me know what you think about it.

**Comment by Emmanueldmlr** [comment]
What we currently have worked though.

**Comment by Emmanueldmlr** [comment]
> can you elaborate the reason? I think w current graph you can get top 10 order by earnings to show the UI of all time.
> 
> >

I have implemented this. This is working fine.

**Comment by Emmanueldmlr** [comment]
> yes, for now what we can do its to query the current graph to see if he has rewards. If he has rewards, we can query the top 3 to see if it needs a trophy icon or a star icon

This works then... @OleanjiKingCode, this is the update you need to make to the graph.

**Comment by Emmanueldmlr** [comment]
@OleanjiKingCode, let me know when you are done. @kesar, pending the time @OleanjiKingCode is working on this, is there any other issue I can jump on?

**Comment by kesar** [comment]
@Emmanueldmlr 

{
  editors(where: {Address}) {
    TotalRewards
  }
}

using where in editors you can get if the user has result for his profile
and using editors sorted by total rewards limit 3 you can see if he is in top 3

with that you should have all that you need for the v1

**Comment by Emmanueldmlr** [comment]
Nice, I will do this. Thanks

**Comment by Emmanueldmlr** [comment]
image

File: src/data/LeaderBoardData.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+export const LEADERBOARD_DATA = [
```
Comment (kesar) [inline src/data/LeaderBoardData.ts:1]: this can be removed if its already using API

**Comment by Emmanueldmlr** [comment]
removed now.

**Review by kesar — APPROVED** [review]

**Comment by kesar** [comment]
merge and add env vars in vercel for dev and prod

**Pull request merged**